src/components/Projects/WBS/WBSDetail/AddTask/components/TagsSearch.jsx

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable function-paren-newline */
+/* eslint-disable implicit-arrow-linebreak */
+/* eslint-disable arrow-parens */
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-param-reassign */
+/* eslint-disable react/prop-types */
+/* eslint-disable max-len */
+/* eslint-disable react/no-array-index-key */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+import React, { useState } from 'react';
+import TagSent from './TagSent';
+import './TagsSearch.css';
+
+function TagsSearch({ data, placeholder }) {
+  const [tags, setTags] = useState([]);
+  const [isHidden, setIsHidden] = useState(false);
+  const [filteredData, setFilteredData] = useState([]);
+  const [selectedIndex, setSelectedIndex] = useState(0);
+
+  // Se tiver como usar a tag aqui no lugar de indexToRemove seria melhor
+  // Pq se tiver outro component com o msm numero eles podem ser filtrados ao mesmo tempo
+  const removeTags = indexToRemove => {
+    setTags(tags.filter((_, index) => index !== indexToRemove));
+  };
+
+  const addTags = event => {
+    if (event.key === 'Enter' && event.target.value !== '') {
+      setTags([...tags, event.target.value]);
+      event.target.value = '';
+    }
+  };
+
+  const handleClick = event => {
+    setTags([...tags, event.target.innerText]);
+    event.target.value = '';
+    setIsHidden(!isHidden);
+  };
+
+  const handleFilter = event => {
+    const searchWord = event.target.value;
+    const noRepeatTags = data.filter(e => !tags.includes(e.name));
+    const newFilter = noRepeatTags.filter(value =>
```

Comment:
I think this will work. 
```
 
array.filter((element) => {
          return array.name.toLocaleLowerCase().includes(searchResource);
        })
```

if this doesn't work by just plugging in existing code you can figure it out.